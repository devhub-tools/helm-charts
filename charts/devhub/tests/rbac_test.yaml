suite: Secret
templates:
  - rbac.yaml
release:
  name: devhub
  namespace: devhub

tests:
  - it: should render rbac
    asserts:
      - matchSnapshot: {}
