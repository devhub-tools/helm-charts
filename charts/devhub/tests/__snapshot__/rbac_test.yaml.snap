should render rbac:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        helm.sh/chart: devhub-2.1.0
      name: devhub
    rules:
      - apiGroups:
          - batch
        resources:
          - jobs
        verbs:
          - create
          - delete
      - apiGroups:
          - ""
        resources:
          - pods
          - pods/log
        verbs:
          - list
          - get
      - apiGroups:
          - ""
        resources:
          - secrets
          - serviceaccounts/token
        verbs:
          - create
          - update
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      labels:
        app.kubernetes.io/instance: devhub
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: devhub
        helm.sh/chart: devhub-2.1.0
      name: devhub
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: devhub
    subjects:
      - kind: ServiceAccount
        name: devhub
        namespace: devhub
